### DISCUSSION TITLE
Can Fan Control manage the Aerocool Mirage L240 AIO?

### DISCUSSION THREAD

**Discussion by JoseHidalgo** [Rem0o/FanControl.Releases#2106] [General]
Hi everybody,

Here's the product page: https://aerocool.io/product/mirage-l240/

Ideally I'd like to manage the AIO fan speeds and also the pump speed.
Can it be done? Do I need a specific plugin?

Thanks in advance!

![Mirage-L240-Infographic700x700px-01](https://github.com/Rem0o/FanControl.Releases/assets/10851603/069a7e9c-6bb7-42aa-943b-a989a27de9fb)

**Comment by JoseHidalgo** [comment]
... hello ?... Somebody please ?

**Reply by Rem0o** [reply]
If you plug it using 4 pins/3pins normal fan plugs into your motherboard, FanControl can control it. If it's plugged via a USB header and has manufacturer software to control it, then a plugin with the HID device protocol needs to be developped.

**Reply by JoseHidalgo** [reply]
Thank you so much @Rem0o for clarifying how FanControl works. :)

Aerocool doesn't have any manufacturer software. For the aforementioned AIO, RGB connectors aside:
- The pump is powered via SATA, the speed signal is provided via a 3-pin fan connection.
- The fans are operated and controlled via 4-pin PWM connections, so I'll be able to control them.

This quote says that I should connect the pump to the AIO_PUMP connector instead of the CPU_FAN / CPU_OPT connectors:

> " The Pump will be less prone to wear and tear if it runs at a static speed. This is why there is an AIO_PUMP or PUMP_HEADER on many Motherboards. You can plug in your Pump (if it comes with a separate 3-Pin connector) to this Header to control it separately from the radiator’s fans or set it to a fixed 50% to 100% PWM speed. "

Source: https://www.cgdirector.com/connect-aio-pump/

My motherboard BIOS confirms that the AIO_PUMP can detect the type of installed pump and control it in 3-pin (DC) or 4-pin (PWM), via a variety of predefined profiles, including a manual one.

So I guess all that should also be manageable by FanControl, right?

![2023 10 29 - 14 48 11](https://github.com/Rem0o/FanControl.Releases/assets/10851603/0fb04078-52d2-4c64-abe6-b4cd9d795889)

![2023 10 29 - 14 53 38](https://github.com/Rem0o/FanControl.Releases/assets/10851603/b98325e9-47b6-4316-970e-2d1a7f2d3322)
